doc/pacignore.8.md

```diff
@@ -0,0 +1,66 @@
+% PACIGNORE(8) User Manual
+
+# USAGE
+
+**pacignore** ls [option...]\
+**pacignore** {check|add|rm} [option...] \<pkg>\ [pkg...]
+
+# DESCRIPTION
+
+Ignore Arch Linux packages.
+
+# SUBCOMMANDS
+
+**ls** _\<subcommand\>_\
+
+> List package(s) in the IgnorePkg directive.
+
+**check** _\<subcommand\>_\
+
+> Check if package(s) are included in the IgnorePkg directive.
+
+**add** _\<subcommand\>_\
+
+> Add package(s) to the IgnorePkg directive.
+
+**rm** _\<subcommand\>_\
+
+> Remove package(s) from the IgnorePkg directive.
+
+# OPTIONS
+
+**\-c** _\<path\>_\
+
+> Pacman configuration file, default is _/etc/pacman.conf_.
+
+**\-h**\ _\<flag\>_\
```

Comment:
Added a mandoc for `pacignore`.

Question regarding syntax:

CLI options such as `-h` don't require an argument, so they should not require an argument type enclosed by the `<*>` characters. However, `-h` renders quite strange in the mandoc without an argument due to lots of space between it and the description below. So I added a "phantom" argument called `<flag>` to indicate this is just a flag and nothing more.

Is it fine if we do it this way (also for other similar cases)?

Reply:
> So I added a "phantom" argument ... this is just a flag

Hmm. I think it implies `-h` "takes a flag", whatever that might mean. So it feels confusing to me.

> -h renders quite strange in the mandoc without an argument due to lots of space between it and the description below

I did a small test in another project and I see,

![](https://files.pbrisbin.com/screenshots/screenshot.89171.png)

This seems totally fine to me. Sure there's some space, but it's all consistent.

That said, you could certainly do something like,

```markdown
**\-h** Show help message
```

And get

![](https://files.pbrisbin.com/screenshots/screenshot.89661.png)

:shrug: